python/wiki_etl.py

```diff
@@ -0,0 +1,417 @@
+                                     self.xmljson_mapper_mb,
+                                     self.xmljson_mapper_mb_heap,
+                                     self.xmljson_reducer_mb,
+                                     self.xmljson_reducer_mb_heap,
+                                     self.force,
+                                     self.debug)
+        converter.run()
+
+    def _hive_create_fulltext_table(self):
+        logger.debug("Launchung HQLrunner for fulltext table creation")
+        hql_path = self._hive_path(HQL_SCRIPT_FULLTEXT_TABLE)
+        hql_params = [
+            self._hive_param(HQL_PARAM_HCATALOG_PATH, self.hive_hcatalog),
+            self._hive_param(HQL_PARAM_FULLTEXT_TABLE, self.fulltext_table),
+            self._hive_param(HQL_PARAM_DATA_PATH, self.json_path)
+        ]
+        hql_runner = HQLRunner(hql_path,
+                               hql_params,
+                               self.hive_server,
+                               self.hive_port,
+                               self.user,
+                               self.hive_database,
+                               self.debug)
+        hql_runner.run()
+
+    def _hive_create_metadata_table(self):
+        logger.debug("Launching HQLrunner for metadata table creation")
+        hql_path = self._hive_path(HQL_SCRIPT_METADATA_TABLE)
+        hql_params = [
+            self._hive_param(HQL_PARAM_METADATA_TABLE, self.metadata_table),
+            self._hive_param(HQL_PARAM_DATA_PATH, self.meta_path)
+        ]
+        hql_runner = HQLRunner(hql_path,
+                               hql_params,
+                               self.hive_server,
+                               self.hive_port,
+                               self.user,
+                               self.hive_database,
+                               self.debug)
+        hql_runner.run()
+
+    def _hive_load_metadata_table(self):
+        logger.debug("Launching HQLrunner loading metadata table")
+        hql_path = self._hive_path(HQL_SCRIPT_METADATA_LOAD)
+        hql_params = [
+            self._hive_param(HQL_PARAM_HCATALOG_PATH, self.hive_hcatalog),
+            self._hive_param(HQL_PARAM_QUEUE, self.queue),
+            self._hive_param(HQL_PARAM_REDUCERS, self.hive_metadata_reducers),
+            self._hive_param(HQL_PARAM_COMPRESSION,
+                             self.hive_metadata_compression),
+            self._hive_param(HQL_PARAM_METADATA_TABLE, self.metadata_table),
+            self._hive_param(HQL_PARAM_FULLTEXT_TABLE, self.fulltext_table)
+        ]
+        hql_runner = HQLRunner(hql_path,
+                               hql_params,
+                               self.hive_server,
+                               self.hive_port,
+                               self.user,
+                               self.hive_database,
+                               self.debug)
+        hql_runner.run()
+
+    def run(self):
+        if not self.no_download:
```

Comment:
I'm no seeing where we are using `self.download_success_flag`. It would make sense to me to check it here and either log a warning or throw an error if we are trying to move on without a successful download. Both for the case that `self._download()` does not finish perfectly and when `self.no_download == True`